src/libos/src/fs/channel.rs

```diff
@@ -77,12 +77,32 @@ impl<I> Channel<I> {
         (producer, consumer)
     }
 
+    pub fn capacity(&self) -> usize {
+        self.consumer.capacity()
+    }
+
     pub fn items_to_consume(&self) -> usize {
         self.consumer.items_to_consume()
     }
 
+    pub fn consumer_nonblocking(&self) -> bool {
+        self.consumer.is_nonblocking.load(Ordering::Acquire)
+    }
+
+    pub fn producer_nonblocking(&self) -> bool {
+        self.producer.is_nonblocking.load(Ordering::Acquire)
+    }
+
     pub fn set_nonblocking(&self, nonblocking: bool) {
+        self.set_consumer_nonblocking(nonblocking);
+        self.set_producer_nonblocking(nonblocking);
+    }
+
+    pub fn set_consumer_nonblocking(&self, nonblocking: bool) {
         self.consumer.set_nonblocking(nonblocking);
+    }
+
+    pub fn set_producer_nonblocking(&self, nonblocking: bool) {
```

Comment:
Exposing these APIs is a bit of ugly. I think a more elegant solution is add `consumer` and `producer` methods so that the user of `Channel` can access them directly.